File: SoundMexPro/tutorialpy/t_00b_init_tutorial.py
#
#                    Example for usage of SoundMexPro
# Copyright Hoerzentrum Oldenburg gGmbH Oldenburg 2021, written by Daniel Berg
#
# Helper script for the SoundMexPro tutorials. Called at the beginning of
# every tutorial script. Reads inifile t_smpcfg.ini and returns content
 
import configparser
import os

def smp_get_cfg(inputs):
    config = configparser.ConfigParser()
    config.read(os.path.dirname(__file__) + '\\t_smpcfg.ini')
    retvals = dict(config.items('settings'))
    # remove inputs if requested
    if inputs == 0:
        retvals.pop("input")    
    return retvals
    
    

File: SoundMexPro/tutorial/t_00b_init_tutorial.m
% 
%        Initialization script for the SoundMexPro tutorial
% Copyright Hoerzentrum Oldenburg gGmbH Oldenburg 2015, written by Daniel Berg
% 
% Helper script for the SoundMexPro tutorials. Called at the beginning of
% every tutorial script. Loads configuration from t_smpcfg.mat, or - if it
% doesn't exist - calls t_00_setup_tutorial.m to create it
%

clc;
close all;
clear soundmexpro;



% create full filename of confirguration file
[folder, name, ext] = fileparts(mfilename('fullpath'));
cfgfile = [folder '\t_smpcfg.mat'];

% load structure smpcfg from file containing driver, input and output
% channels to be used in tutorial.

% if it doesn't exist, call script t_00_setup.m to create it
if ~exist(cfgfile, 'file')
   t_00_setup_tutorial;
   if ~exist(cfgfile, 'file')
       return;
   end
end

load(cfgfile);
